carbon-design-system/carbon-addons-iot-react
# fix(table): preserve whitespace in cells with multiple spaces

Closes #3011 

**Summary**

- adds a class setting `white-space: pre` in table cells that have multiple continuous spaces, so that whitespace is preserved in those cells.

**Change List (commits, features, bugs, etc)**

- add `iot--table__cell-text--preserve` class
- set class in TableCellRenderer when preserveCellWhiteSpace prop is true
- add tests to confirm

**Acceptance Test (how to verify the PR)**

- Go to the table with full rowEdit example
- turn on preserveCellWhiteSpace knob
- edit the first row and add numbers with multiple spaces in the number column
- save
- confirm your multiple spaces are still visible in the output and whitespace is preserved.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- no adverse effects on other strings



**Things to look for during review**
